### PR TITLE
Remove TopDown engine from repository

### LINKED ISSUE #1112: [REFACTOR] Remove TD engine from the repository
1. Remove TopDownEngine from the repository,
2. The documentation must also be updated so that every new user knows they should download and include the engine package in the 'Assets' folder (and must purchase the license).

### PR DESCRIPTION
**Pull request by tkz00**
Closes #1112 

## Motivation & Summary of Changes

Since TopDown Engine is an external dependency we removed it from the repository.

## How to test it?

To accurately test this PR it's necessary to backup the TopDown Engine folder, located in `/client/Assets/ThirdParty` while on the main branch, then switch to the PR's branch and manually add the engine's folder in the corresponding folder of the client, after that you should verify no problems arise in Unity and that the game can be played as expected.

## Checklist
- [X] I have tested the changes locally.
- [X] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [X] This change requires new documentation.
  - [X] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Comment by manucamejo** [comment]
Nice work @tkz00, I have some questions:
- How can I run from scratch if I don't have TopDownEngine installed? Did you try it? Can you document it?
- Is it going to crash when we merge it? We have to do something?

**Comment by tkz00** [comment]
> Nice work @tkz00, I have some questions:
> 
>     * How can I run from scratch if I don't have TopDownEngine installed? Did you try it? Can you document it?
> 
>     * Is it going to crash when we merge it? We have to do something?

Everyone on the team should have no problems running and working on the game when we merge, since the TopDownEngine folder was removed from the repository and added to the gitignore so it will not affect the "local" TopDownEngine folders we have. If someone clones the project from scratch he will need to add the engine's folder to the correct location inside the project. This has been specified in the readme, but take a look at it and let me know if the instructions look clear or if you would change or add something!

**Comment by BertovDev** [comment]
> > Nice work @tkz00, I have some questions:
> > ```
> > * How can I run from scratch if I don't have TopDownEngine installed? Did you try it? Can you document it?
> > 
> > * Is it going to crash when we merge it? We have to do something?
> > ```
> 
> Everyone on the team should have no problems running and working on the game when we merge, since the TopDownEngine folder was removed from the repository and added to the gitignore so it will not affect the "local" TopDownEngine folders we have. If someone clones the project from scratch he will need to add the engine's folder to the correct location inside the project. This has been specified in the readme, but take a look at it and let me know if the instructions look clear or if you would change or add something!

IMO, further documentation should be added on how to setup the engine, something like:

- Where it is going to be placed the engine to be downloaded?

File: README.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ make setup
 ## Setup Unity
 - On Unity Hub click on the add project button and select the `client` folder.
 - Select the correct version of the editor, by default it will show the version needed for the project but you need to select if you want to download the Intel or Silicon version.
+- It's also needed to download the [Top Down Engine](https://assetstore.unity.com/packages/templates/systems/topdown-engine-89636) by [More Mountains](https://moremountains.com) and include it in the assets folder. For this it's necessary to purchase the license of the engine.
```
Comment (BertovDev) [inline README.md:33]: I believe you should include in the asset folder but inside ThirdParty

`Assets/ThirdParty`

Comment (tkz00) [reply README.md:33]: Nice catch, I'll modify the readme.

**Review by ncontinanza — CHANGES_REQUESTED** [review]
Please elaborate a bit more on how to import the TopDown engine folder for testing this PR

**Comment by tkz00** [comment]
> Please elaborate a bit more on how to import the TopDown engine folder for testing this PR

Already expanded the documentation!

**Review by BertovDev — APPROVED** [review]
Nice

**Review by ncontinanza — APPROVED** [review]
Working good! Tested it locally and in an Android device

**Pull request merged**